core/src/cats/sql_get.cc

```diff
@@ -1293,6 +1293,42 @@ bool BareosDb::GetUsedBaseJobids(JobControlRecord* jcr,
   return SqlQueryWithHandler(query.c_str(), DbListHandler, result);
 }
 
+/*
+ * Remove all jobs in filelist from job table that have no files
+ *  - The job never had any files
+ *  - The job files were purged or pruned
+ */
+bool BareosDb::DeleteZeroFileJobsinList(JobControlRecord* jcr,
+                                        JobDbRecord* jr,
+                                        db_list_ctx* jobids)
+{
+  db_list_ctx deleted_jobids;
+  std::string query{
+      "DELETE FROM Job where (JobFiles=0 OR PurgedFiles=1) AND JobID in ("};
```

Comment:
do we really want to implicitly purge jobs when their files were purged?
I don't think that's right.

Reply:
also, just doing an SQL `DELETE` will potentially leave stale records in tables JobMedia, Log, basefiles, PathVisibility, NDMPJobEnvironment, JobStats and RestoreObject.

Reply:
I guess we need to determine the list and then do whatever"delete jobid=XXX" would be doing.